# HG changeset patch
# User -r124899
# Date 1298504194 28800
# Branch pnacl-sfi
# Node ID 0337a0cbbcaf3199d3585caa58099601525fad8f
# Parent a7a7bb9fbd8d298f53d88509072db7475db98a49
Just what it says - for ARM/MC/ELF
http://llvm.org/viewvc/llvm-project?view=rev&revision=124899


Teach ARM/MC/ELF about EF_ARM_EABI_VERSION. The magic number is set to
5 to match the current doc.
Added FIXME reminder Make it really configurable later.
NOTE:
NaCl requires an special ELF e_flags. So does ARM.
PNaCl lib/MC/ELFObjectwriter.cpp had a @LOCALMOD to deal with the NaCl specific e_flags which conflicted with the necessary update to ARM. This required the merged version to have the @LOCALMOD@ as well to handle both cases.

 From llvm-pnacl-0006-226-0337a0cbbcaf3199d3585caa58099601525fad8f.patch

 From .hg/patches/llvm-pnacl-0006-226-0337a0cbbcaf3199d3585caa58099601525fad8f.patch.stripped

diff -r a7a7bb9fbd8d llvm-trunk/lib/MC/ELFObjectWriter.cpp
--- a/llvm-trunk/lib/MC/ELFObjectWriter.cpp	Tue Feb 22 17:59:22 2011 -0500
+++ b/llvm-trunk/lib/MC/ELFObjectWriter.cpp	Mon Jul 11 09:35:45 2011 -0700
@@ -1248,7 +1248,16 @@
 
 // FIXME: get the real EABI Version from the Triple.
 void ARMELFObjectWriter::WriteEFlags() {
-  Write32(ELF::EF_ARM_EABIMASK & DefaultEABIVersion);
+  // @LOCALMOD-BEGIN
+  unsigned e_flag = ELF::EF_ARM_EABIMASK & DefaultEABIVersion;
+  switch (TargetObjectWriter->getOSType()) {
+  case Triple::NativeClient:
+    e_flag |= ELF::EF_NACL_ALIGN_32;
+    break;
+  default: break;
+  }
+  Write32(e_flag);
+  // @LOCALMOD-END
 }
 
 // In ARM, _MergedGlobals and other most symbols get emitted directly.
